sdk/tables/azure-data-tables/azure/data/tables/aio/_table_batch_async.py

```diff
@@ -490,7 +481,7 @@ def _batch_delete_entity(
         table: str,
         partition_key: str,
         row_key: str,
-        if_match: str,
+        if_match: Optional[str],
```

Comment:
This isn't `Optional`. This value must be either `"*"` or an etag.

Reply:
It is, will not set "If-Match" in request header when if_match is None. 
I'll leave _table_batch.py and _table_batch_async.py as they are since you already did updates in the batch refactor PR.